protocol/xmrmaker/message_handler.go

```diff
@@ -226,6 +201,40 @@ func (s *swapState) handleNotifyETHLocked(msg *message.NotifyETHLocked) (net.Mes
 	return out, nil
 }
 
+func (s *swapState) handleT0Expired() {
+	doExit := false
+	s.lockState()
+	defer func() {
+		s.unlockState()
+		if doExit {
+			_ = s.Exit()
+		}
+	}()
+
+	if !s.info.Status().IsOngoing() {
+		// swap was already completed, just return
+		return
+	}
+
+	// we can now call Claim()
+	txHash, err := s.claimFunds()
+	if err != nil {
+		log.Errorf("failed to claim: err=%s", err)
+		// TODO: retry claim, depending on error
+		doExit = true
```

Comment:
@noot Based on the comments I left above, I can change this to call `s.exit()`, which does not grab the lock, here and then simplify the `defer` code at the top.